test/tools/fuzzTesting/fuzzHelper.cpp

```diff
@@ -509,19 +511,49 @@ void RandomCodeOptions::setWeight(eth::Instruction _opCode, int _weight)
 	setWeights();
 }
 
-void RandomCodeOptions::addAddress(Address const& _address)
+void RandomCodeOptions::addAddress(Address const& _address, AddressType _type)
 {
-	addressList.push_back(_address);
+	switch(_type)
+	{
+		case AddressType::CALLONLY:
+			callAddressList.push_back(_address);
+			break;
+		case AddressType::ALL:
+		case AddressType::ACCOUNT:
+			stateAddressList.push_back(_address);
+			callAddressList.push_back(_address);
+			break;
+		default:
+			BOOST_ERROR("RandomCodeOptions::addAddress: Unexpected AddressType!");
+		break;
+	}
 }
 
-Address RandomCodeOptions::getRandomAddress() const
+Address RandomCodeOptions::getRandomAddress(AddressType _type) const
 {
-	if (addressList.size() > 0)
+	switch(_type)
 	{
-		int index = (int)RandomCode::randomUniInt() % addressList.size();
-		return addressList[index];
+		case AddressType::CALLONLY:
+			if (RandomCode::randomPercent() < emptyAddressProbability)
+				return ZeroAddress;
+			return callAddressList[(int)RandomCode::randomUniInt(0, callAddressList.size())];
```

Comment:
then it would be that question why callOlnly request  return  both inState and CallOnly